rpmlint/checks/ExecDocCheck.py

```diff
@@ -0,0 +1,44 @@
+from pathlib import Path
+import stat
+
+from rpmlint.checks.AbstractCheck import AbstractCheck
+
+
+class ExecDocCheck(AbstractCheck):
+    @staticmethod
+    def ignore_pkg(name):
+        if name.startswith('bundle-') or '-devel' in name or '-doc' in name:
+            return True
+        return False
+
+    def check(self, pkg):
+        if pkg.isSource():
+            return
+
+        files = pkg.files()
+        complete_size = 0
+        for _, pkgfile in files.items():
+            if stat.S_ISREG(pkgfile.mode):
+                complete_size += pkgfile.size
+
+        doc_size = 0
+        for f in pkg.docFiles():
+            if stat.S_ISREG(files[f].mode):
+                doc_size += files[f].size
+
+        if doc_size * 2 >= complete_size and doc_size > 100 * 1024 and not self.ignore_pkg(pkg.name):
+            self.output.add_info('W', pkg, 'package-with-huge-docs: %3d%%' % (doc_size * 100 / complete_size))
```

Comment:
This whole calculation above starting with line 18 should be separate function which gets called if not ignoring the package.

Reply:
Still not what I ment, now from line 22 form this file until line 34 this hould be separate function only called when self.ignore_pkg for it is false.